boot/bootutil/include/bootutil/bootutil_log.h

```diff
@@ -135,16 +135,21 @@ int sim_log_enabled(int level);
  */
 #elif defined(MCUBOOT_MYNEWT)
 
+#include <mcuboot_config/mcuboot_config.h>
```

Comment:
The switch to Kconfig (https://github.com/runtimeco/mcuboot/pull/272) was careful to make sure all the mcuboot_config.h options are documented in the template file, and that they don't depend on any other macros/values in the system. That is, bootutil depends on mcuboot_config.h, and not the other way around.

This PR has two problems from that perspective:

- It adds MCUBOOT_LOG_LEVEL without documentation in the template file
- The new macro gets its values from macros (BOOT_LOG_LEVEL_xxx) defined in bootutil

Can this PR please be reworked to manage the dependencies in a cleaner way?

I will add a follow-up with a design I considered in #272 but postponed due to lack of time.

Reply:
> I will add a follow-up with a design I considered in #272 but postponed due to lack of time.

Here is what I was thinking:

- mcuboot_config.h would have a single additional config option added and documented, named `MCUBOOT_HAVE_LOGGING`
- If that option is defined, then the platform guarantees that bootutil can `#include mcuboot_config/mcuboot_logging.h`; otherwise, that file may not be included
- If mcuboot_logging.h is present, it must:
  - define MCUBOOT_LOG_LEVEL_{OFF,ERROR,...DEBUG} appropriately for the platform
  - detect the value of MCUBOOT_LOG_LEVEL if it is defined, and give it a sane platform-specific default if it is undefined
  - define MCUBOOT_LOG_{ERR,WRN,...} appropriately for the platform and the current MCUBOOT_LOG_LEVEL
- This file (bootutil_log.h) can then include mcuboot_config.h and check `MCUBOOT_HAVE_LOGGING`
   - if MCUBOOT_HAVE_LOGGING is defined, use the platform definitions, as in: `#define BOOT_LOG_LEVEL_{OFF,ERROR,...} MCUBOOT_LOG_LEVEL_{OFF,ERROR,...}` etc., and `#define BOOT_LOG_{ERR,WRN,...} MCUBOOT_LOG_{ERR,WRN,...}`, etc.
   - if absent, use stub implementations which ignore their arguments
- The rest of the library can then just include bootutil.h and use BOOT_LOG_xxx safely without worrying about the details.